controllers/front/landing.php

```diff
@@ -105,11 +111,14 @@ private function renderProductList()
 
     private function renderProductList16()
     {
+        $products_length = count($this->products);
+        for ($i = 0; $i < $products_length; ++$i) {
+            $this->landing_data['blocks'][$i]['products'] = $this->products[$i];
+        }
```

Comment:
podemos quitar aquí este for y hacerlo como en el bloque de renderProductList() ?

Reply:
De hecho creo que con la modificación que me has sugerido antes con quitarlo simplemente bastaría. Lo he probado en el Prestashop 1.6 que tengo y parece que funciona tal y como esperaba